### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by mohammed-sas**
1.added wishlist page
2.products can be removed from wishlist or can be moved to cart in the wishlist page
3.if a product already exists in cart then moving product from wishlist would increment the quantity as well removes the product from the wishlist
[deploy preview](https://deploy-preview-7--mhd-racquets-store.netlify.app/)

### CODE REVIEW

File: src/context/wishlist-context.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                authorization:token
+            }});
+            wishlistDispatch({type:"DELETE_FROM_WISHLIST",payload:response.data.wishlist});
+        }catch(error){
+            console.log(error);
+        }
+    }
+
+    return {wishlistState,wishlistDispatch,addToWishlist,deleteFromWishlist};
+}
```
Comment (starver20) [inline src/context/wishlist-context.js]: Very well written code.

File: src/context/wishlist-context.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import axios from "axios";
+import { createContext, useContext, useReducer } from "react";
+import { wishlistReducer } from "../reducer/wishlistReducer";
+
+const initialState = {
+    wishlist:[],
+    totalItems:0
```
Comment (starver20) [inline src/context/wishlist-context.js:7]: You can find the total number of items on the fly in whichever component you want using wishlist.length . I dont think you need to store it seperately here.

File: src/reducer/wishlistReducer.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                ...state,
+                wishlist:[...action.payload],
+                totalItems:action.payload.length
+            };
+        case "DELETE_FROM_WISHLIST":
+            return{
+                ...state,
+                wishlist:[...action.payload],
+                totalItems:action.payload.length
+            }
```
Comment (starver20) [inline src/reducer/wishlistReducer.js:15]: You can try to combine these both into one function, where you check if the item already exits then you remove it, or else you add it to wishlist.

**Review by starver20 — APPROVED** [review]
Nicely written code @mohammed-sas. Suggested couple of suggestions.

File: src/reducer/wishlistReducer.js (hunk 3 above)
Comment (mohammed-sas) [reply src/reducer/wishlistReducer.js:15]: for readability purpose i've seperated them

File: src/context/wishlist-context.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+const useWishlistAPI=()=>{
+    const [wishlistState,wishlistDispatch] = useReducer(wishlistReducer,initialState);
+    let token = localStorage.getItem("token");
+    const addToWishlist=async (product)=>{
+        try{
+            let response = await axios.post("/api/user/wishlist",{product},{headers:{
+                authorization:token
+            }});
+            wishlistDispatch({type:"ADD_TO_WISHLIST",payload:response.data.wishlist})
```
Comment (AtulPant2704) [inline src/context/wishlist-context.js:28]: Check response.status over here, and if it is 200 then send dispatch, otherwise send error.

**Review by AtulPant2704 — APPROVED** [review]
Other than checking response.status, rest all looks good.

File: src/context/wishlist-context.js (hunk 4 above)
Comment (mohammed-sas) [reply src/context/wishlist-context.js:28]: Sure

**Pull request merged**